homeassistant/components/dsmr/config_flow.py

```diff
@@ -55,40 +54,40 @@ def __init__(
         self._port = port
         self._dsmr_version = dsmr_version
         self._protocol = protocol
-        self._telegram: dict[str, DSMRObject] = {}
-        self._equipment_identifier = obis_ref.EQUIPMENT_IDENTIFIER
+        self._telegram: Telegram = None
+        self._equipment_identifier = "EQUIPMENT_IDENTIFIER"
         if dsmr_version == "5B":
-            self._equipment_identifier = obis_ref.BELGIUM_EQUIPMENT_IDENTIFIER
+            self._equipment_identifier = "BELGIUM_EQUIPMENT_IDENTIFIER"
         if dsmr_version == "5L":
-            self._equipment_identifier = obis_ref.LUXEMBOURG_EQUIPMENT_IDENTIFIER
+            self._equipment_identifier = "LUXEMBOURG_EQUIPMENT_IDENTIFIER"
         if dsmr_version == "Q3D":
-            self._equipment_identifier = obis_ref.Q3D_EQUIPMENT_IDENTIFIER
+            self._equipment_identifier = "Q3D_EQUIPMENT_IDENTIFIER"
 
     def equipment_identifier(self) -> str | None:
         """Equipment identifier."""
-        if self._equipment_identifier in self._telegram:
-            dsmr_object = self._telegram[self._equipment_identifier]
+        if hasattr(self._telegram, self._equipment_identifier):
+            dsmr_object = getattr(self._telegram, self._equipment_identifier)
             identifier: str | None = getattr(dsmr_object, "value", None)
             return identifier
         return None
 
     def equipment_identifier_gas(self) -> str | None:
         """Equipment identifier gas."""
-        if obis_ref.EQUIPMENT_IDENTIFIER_GAS in self._telegram:
-            dsmr_object = self._telegram[obis_ref.EQUIPMENT_IDENTIFIER_GAS]
+        if hasattr(self._telegram, "EQUIPMENT_IDENTIFIER_GAS"):
+            dsmr_object = getattr(self._telegram, "EQUIPMENT_IDENTIFIER_GAS")
```

Comment:
Line 77 can indeed be changed to `self._telegram.EQUIPMENT_IDENTIFIER_GAS`, that's correct.
But in the other cases it can't.

And why this change? Cause the dict interface from dsmr_parser is deprecated, so if it gets removed, the code will keep working.
But next to that, if we want to change to v1.4.0 of dsmr_parser to include the new way to access mbus devices, we can't access the mbus devices anymore using the dict.

Reply:
`self._telegram.EQUIPMENT_IDENTIFIER_GAS` <- that is not a correct naming for a Python variable/property either.

I'm not sure what is going on upstream, we are either using it wrong or they are doing it wrong.

Reply:
As the attributes are dynamic depending on the DSMR Telegram that is parsed, it was decided in dsmr_parser to just set attributes when parsing the Telegram.
And the way to access the values is described here: https://github.com/ndokter/dsmr_parser/blob/master/README.rst?plain=1#L233

The way it happens in dsmr_parser can be reviewed, but this is how it works for now.

Reply:
Not my choice :)
In this line `self._telegram.EQUIPMENT_IDENTIFIER_GAS` can be used.